plugins/metalink/utils.c

```diff
@@ -720,128 +720,171 @@ TDNFParseUrlTag(
+        elementInfo->dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(elementInfo->dwError);
     }
 
-    //Looping through all the nodes from root and parse all children nodes.
-    while(node)
+    if(!strcmp(elementInfo->startElement, TAG_NAME_FILE))
+    {
+        elementInfo->dwError = TDNFParseFileTag(userData);
+        BAIL_ON_TDNF_ERROR(elementInfo->dwError);
+    }
+    else if(!strcmp(elementInfo->startElement, TAG_NAME_SIZE))
     {
-        if(node->type == XML_ELEMENT_NODE)
+        //Get File Size.
+        TDNFAllocateStringN(val, len, &size);
+
+        if(!size)
         {
-            if(!strcmp((const char*)node->name, TAG_NAME_FILE))
-            {
-                dwError = TDNFParseFileTag(ml_ctx, node, filename);
-                BAIL_ON_TDNF_ERROR(dwError);
-            }
-            else if(!strcmp((const char*)node->name, TAG_NAME_SIZE))
-            {
-                //Get File Size.
-                xmlContValue = xmlNodeGetContent(node);
-                if(!xmlContValue)
-                {
-                    dwError = ERROR_TDNF_METALINK_PARSER_MISSING_FILE_SIZE;
-                    pr_err("XML Parser Error:File size is missing: %s", size);
-                    BAIL_ON_TDNF_ERROR(dwError);
-                }
-                size = (char*)xmlContValue;
-                if(sscanf(size, "%ld", &(ml_ctx->size)) != 1)
-                {
-                    dwError = ERROR_TDNF_INVALID_PARAMETER;
-                    pr_err("XML Parser Warning: size is invalid value: %s\n", size);
-                    BAIL_ON_TDNF_ERROR(dwError);
-                }
-            }
-            else if(!strcmp((const char*)node->name, TAG_NAME_HASH))
-            {
-                dwError = TDNFParseHashTag(ml_ctx, node);
-                BAIL_ON_TDNF_ERROR(dwError);
-            }
-            else if(!strcmp((const char*)node->name, TAG_NAME_URL))
-            {
-                dwError = TDNFParseUrlTag(ml_ctx, node);
-                BAIL_ON_TDNF_ERROR(dwError);
-            }
+            elementInfo->dwError = ERROR_TDNF_METALINK_PARSER_MISSING_FILE_SIZE;
+            pr_err("XML Parser Error:File size is missing: %s", size);
+            BAIL_ON_TDNF_ERROR(elementInfo->dwError);
         }
-        if (node->children) {
-            TDNFXmlParseData(ml_ctx, node->children, filename);
+        if(sscanf(size, "%ld", &(elementInfo->ml_ctx->size)) != 1)
+        {
+            elementInfo->dwError = ERROR_TDNF_INVALID_PARAMETER;
+            pr_err("XML Parser Warning: size is invalid value: %s\n", size);
+            BAIL_ON_TDNF_ERROR(elementInfo->dwError);
         }
-        node = node->next;
+    }
+    else if(!strcmp(elementInfo->startElement, TAG_NAME_HASH))
+    {
+        elementInfo->dwError = TDNFParseHashTag(userData, val, len);
+        BAIL_ON_TDNF_ERROR(elementInfo->dwError);
+    }
+    else if(!strcmp(elementInfo->startElement, TAG_NAME_URL) && len > 4)
```

Comment:
Why its 4 not any other number. Better to replace with a meaningful MACRO.

Reply:
Length 4 is used as minimum url length. Added macro for this number